### PR TITLE
[#158104939] re-init sessioninfo and sessioninfoset if token is missing

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#158104939](https://www.pivotaltracker.com/story/show/158104939): Come proxy devo esporre un endpoint con la lista delle sessioni di un utente

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/services/redisSessionStorage.ts [hunk 1]
```diff
@@ -317,6 +295,51 @@ export default class RedisSessionStorage extends RedisStorageUtils
...
+            resolve(
+              this.falsyResponseToError(
+                this.integerReply(err, response),
+                new Error("Error updating user tokens info set")
+              )
+            )
+        );
+      }
+    );
+    return [saveSessionInfo, updateSessionInfoSet];
```
Comment (cloudify) [inline src/services/redisSessionStorage.ts]: if you return multiple promises the caller needs to know how to handle them (Promise.all? Promise.race? how to handle one rejection vs multiple rejections?) -> this leaks the abstraction of a single operation

I suggest you return a single `Promise<Either<Error, boolean>>`

Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: fixed in f5f264b

File: src/services/redisSessionStorage.ts [hunk 2]
```diff
@@ -239,12 +204,23 @@ export default class RedisSessionStorage extends RedisStorageUtils
         );
       }
     );
-    if (isLeft(sessionKeys)) {
+    // tslint:disable-next-line: readonly-array
+    const initizedSessionKey: string[] = [];
```
Comment (cloudify) [inline src/services/redisSessionStorage.ts]: typo

```suggestion
    const initializedSessionKeys: string[] = [];
```

File: src/services/redisSessionStorage.ts [hunk 3]
```diff
@@ -317,6 +293,46 @@ export default class RedisSessionStorage extends RedisStorageUtils
     );
   }
 
+  /*
+  * Return the promises needed to store session info and
```
Comment (cloudify) [inline src/services/redisSessionStorage.ts]:
```suggestion
  * Store session info and update session info set.
  * The returned promise will reject if either operation fail.
```

File: src/services/redisSessionStorage.ts [hunk 4]
```diff
@@ -317,6 +293,46 @@ export default class RedisSessionStorage extends RedisStorageUtils
     );
   }
 
+  /*
+  * Return the promises needed to store session info and 
+  * update session info set
+  */
+  private saveSessionInfo(user: User): Promise<Either<Error, boolean>> {
+    const newSessionInfo: SessionInfo = {
```
Comment (cloudify) [inline src/services/redisSessionStorage.ts]: this method should probably accept just a `SessionInfo`, not a `User` - functions should require the smallest set of information needed

Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: Is required even `user.fiscal_code` at [this](https://github.com/teamdigitale/io-backend/pull/444/files#diff-178a7f5864283f425f4e6d8d513710cfR322) row.

File: src/services/redisSessionStorage.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: fixed in 2037c21

File: src/services/redisSessionStorage.ts (hunk 4 above)
Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: Should I leave the code unchanged? Or I change the function with 2 parameters: `SessionInfo` and `FiscalCode`?

Comment (cloudify) [reply src/services/redisSessionStorage.ts]: yes two params is better

Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: fixed in 1bd3e55

**Pull request merged**